### DISCUSSION TITLE
Is it possible to create a (on prem) file system abstraction with lakeFS & afero?

### DISCUSSION THREAD

**Discussion by jxsl13** [treeverse/lakeFS#2861] [Q&A]
Hi,

I'm wondering whether it might be possible to abstract an on premise filesystem that supports rollback functionality with lakeFS and the interfaces provided by https://github.com/spf13/afero/blob/master/afero.go

**Comment by itaiad200** [comment]
Hey @jxsl13 , that sounds like a really interesting usage for lakeFS, but I'm not sure exactly on what you're trying to achieve. While lakeFS has some of the characteristics of a filesystem, it resembles more to an object-store. What's the use-case you're aiming for and where do you see lakeFS fits in?

**Comment by jxsl13** [comment]
My general idea would be along the lines of this project: https://github.com/jxsl13/backupfs (backup on write filesystem abstraction)
Imagine having a software being provisioned to some target filesystem or target server.
What I'm trying to achieve is to create backups of older versions of the software at the moment when a newer version is being provisioned to the target system.
In case that the new version does have some some problems, I'd like to rollback the software release with the backup filesystem.

Maybe only keep a few working revisions of the software on the target system.


Basically using lakeFS as an embedded version inside of another application in order to backup overwritten software revisions in lakeFS.
